### PR TITLE
system/adb: Replace NuttX special shell service with microADB builtin one

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since the new libuv porting support uv_proccess_t now(https://github.com/apache/incubator-nuttx-apps/pull/1036)

## Impact
Unify the shell implementation

## Testing
sim:adb(https://github.com/apache/incubator-nuttx/pull/5501)

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Comment by spiriou** [comment]
> @spiriou could you take a look?

LGTM.

Now may be the right time to remove config `stm32f103-minimum:adb` as new libuv and adb won't fit in anymore.

**Comment by xiaoxiang781216** [comment]
> > @spiriou could you take a look?
> 
> LGTM.
> 
> Now may be the right time to remove config `stm32f103-minimum:adb` as new libuv and adb won't fit in anymore.

could you give more info why they don't fit together? I would expect the change in libuv and microadb should work with preexisting config.

**Comment by spiriou** [comment]
> > > @spiriou could you take a look?
> > 
> > 
> > LGTM.
> > Now may be the right time to remove config `stm32f103-minimum:adb` as new libuv and adb won't fit in anymore.
> 
> could you give more info why they don't fit together? I would expect the change in libuv and microadb should work with preexisting config.

libuv is optimized for performance at the cost of flash and ram usage. It's fine for most usecases.

stm32f103-minimum board has 20KB of RAM and 64kB of flash.
adb config builds nuttx, usb stack with adb driver, libuv and adb server.

It did barely fit with the reduced footprint port of libuv so it may not work anymore now. I would just drop this config as it's useless anyway. I think the new libuv integration is better for long term purpose as it reduces complexity and makes support and maintenance easier.

**Comment by xiaoxiang781216** [comment]
Ok, understand. Let's see whether the code can fit into the flash. Since I don't have stm32f103-minimum board, the runtime memory usage is hard to check.

**Comment by xiaoxiang781216** [comment]
@spiriou The ci break is fixed here: https://github.com/spiriou/microADB/pull/24

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Seems like those 2 PRs are coupled and CI can't handle it. Let's merge manually.
@xiaoxiang781216 are these 2 changes ready to be merged?

**Comment by xiaoxiang781216** [comment]
> Seems like those 2 PRs are coupled and CI can't handle it. Let's merge manually. @xiaoxiang781216 are these 2 changes ready to be merged?

I update https://github.com/apache/incubator-nuttx/pull/5675, let me try again.

**Comment by xiaoxiang781216** [comment]
> ```
> arm-none-eabi-ld: /github/workspace/sources/nuttx/staging/libapps.a(process-spawn.c.github.workspace.sources.apps.system.libuv.o): in function `uv__process_child_spawn': 
> 149/github/workspace/sources/apps/system/libuv/libuv/src/unix/process-spawn.c:243: undefined reference to `uv_socketpair' 
> 150make[1]: *** [Makefile:152: nuttx] Error 1 
> 151make: *** [tools/Unix.mk:509: nuttx] Error 2 
> 152make: Target 'all' not remade because of errors.
> ```

@pkarashchenko fixed by switch uv_socketpair to uv_pipe.

**Review by pkarashchenko — APPROVED** [review]
